### PR TITLE
raise error in tf.math.polyval for non-list coeffs

### LINKED ISSUE #34947: polyval gives TypeError when run inside tf.function with Tensor coeffs, but not when run eagerly
This is my first issue, so let me know if it is reported in the wrong place. Thanks! 

**System information**
- Have I written custom code: Example Python code to reproduce provided below
- OS Platform and Distribution: Windows 10 Pro 1903
- TensorFlow installed from binary with pip
- TensorFlow version: 'v2.0.0-rc2-26-g64c3d382ca'; '2.0.0'
- Python version: 3.7.4

**Describe the current behavior**
tf.math.polyval works correctly when executing eagerly, but when it is called from within a function that has the @tf.function decorator then a TypeError is raised:

> TypeError: len is not well defined for symbolic Tensors. (eye/diag:0) Please call `x.shape` rather than `len(x)` for shape information.

**Describe the expected behavior**
The output of this example should be the same regardless of eager execution or tf.function decoration:
```
[1 1 5]
[[-2 1 10]]
[[8 2 20]]
[[-18 6 30]]
[[24 24 24]]
```

**Code to reproduce the issue**
```
import tensorflow as tf

x = tf.Variable([[-1.0], [0.0], [1.0]])

@tf.function
def func():
    with tf.GradientTape(persistent=True) as t:
        t.watch(x)
        coeffs = tf.eye(5)
        pv = tf.math.polyval(coeffs, x)
        y = tf.reduce_sum(pv, axis=1)
        dy_dx = t.gradient(y, x)
        d2y_dx2 = t.gradient(dy_dx, x)
        d3y_dx3 = t.gradient(d2y_dx2, x)
        d4y_dx4 = t.gradient(d3y_dx3, x)
    del t

    tf.print(y)
    tf.print(tf.transpose(dy_dx))
    tf.print(tf.transpose(d2y_dx2))
    tf.print(tf.transpose(d3y_dx3))
    tf.print(tf.transpose(d4y_dx4))

func()
```

### PR DESCRIPTION
**Pull request by rushabh-v**
resolves #34947

### CODE REVIEW

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@rushabh-v Can you please fix the build failures? Thanks!

**Comment by rushabh-v** [comment]
Didn't know the system of TF of using `test.Testcase` for tests. corrected the tests. I think it's okay now. @mihaimaruseac Can you please review it again?

File: tensorflow/python/kernel_tests/cwise_ops_test.py [hunk 1]
```diff
@@ -1228,6 +1228,12 @@ def testEmpty(self):
       tf_val = math_ops.polyval(coeffs, x)
       self.assertAllClose(np_val, self.evaluate(tf_val))
 
+  def test_coeffs_raise(self):
+    x = np.random.rand(2, 2).astype(np.float32)
+    coeffs = {}
+    with self.assertRaises(ValueError):
```
Comment (mihaimaruseac) [inline tensorflow/python/kernel_tests/cwise_ops_test.py]: Before you were also checking the error message. I would recommend checking that here too, using method of `tf.TestCase`

Comment (rushabh-v) [reply tensorflow/python/kernel_tests/cwise_ops_test.py]: I have added that. Can you review it again, please?

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**